app/scripts/controllers/network/network-controller.test.js

```diff
@@ -50,224 +69,4106 @@ const PRE_1559_BLOCK = {
+        });
+
+        describe('if the request for eth_getBlockByNumber responds with an error', () => {
+          it('does not update the network details in any way', async () => {
+            await withController(
+              {
+                state: {
+                  provider: {
+                    type: 'rpc',
+                    rpcUrl: 'https://mock-rpc-url',
+                    chainId: '0x1337',
+                  },
+                  networkDetails: {
+                    EIPS: {},
+                  },
+                },
+              },
+              async ({ controller, network }) => {
+                network.mockEssentialRpcCalls({
+                  eth_getBlockByNumber: {
+                    response: {
+                      error: 'oops',
+                    },
+                  },
+                });
+                await withoutCallingLookupNetwork({
+                  controller,
+                  operation: async () => {
+                    await controller.initializeProvider();
+                  },
+                });
+                expect(
+                  controller.store.getState().networkDetails.EIPS['1559'],
+                ).toBeUndefined();
+
+                await waitForStateChanges({
+                  controller,
+                  propertyPath: ['networkDetails'],
+                  count: 0,
+                  operation: async () => {
+                    await controller.lookupNetwork();
+                  },
+                });
+                expect(
+                  controller.store.getState().networkDetails.EIPS['1559'],
+                ).toBeUndefined();
+              },
+            );
+          });
+        });
+      });
+
+      describe('if the request for net_version responds with an error', () => {
+        it('resets the network status to "loading"', async () => {
+          await withController(
+            {
+              state: {
+                provider: {
+                  type: 'rpc',
+                  rpcUrl: 'https://mock-rpc-url',
+                  chainId: '0x1337',
+                },
+              },
+            },
+            async ({ controller, network }) => {
+              network.mockEssentialRpcCalls({
+                net_version: [
+                  {
+                    response: {
+                      result: '42',
+                    },
+                  },
+                  {
+                    response: {
+                      error: 'oops',
+                    },
+                  },
+                ],
+              });
+              await controller.initializeProvider();
+              expect(controller.store.getState().network).toStrictEqual('42');
+
+              await waitForStateChanges({
+                controller,
+                propertyPath: ['network'],
+                operation: async () => {
+                  await controller.lookupNetwork();
+                },
+              });
```

Comment:
Nit: Adding a newline here between the "act" and "assert" steps would be nice:

```suggestion
              });

```